torch/csrc/jit/script/compiler.cpp

```diff
@@ -394,6 +395,36 @@ static inline bool isIntUsedAsIntList(
          *arg.type == *ListType::ofInts() && arg.N;
 }
 
+// Match against against a mutable schema.
+//
+// We need to treat mutable schemas differently because the IR explicitly
+// expresses effects by including a world token in mutable ops. Users do not
+// know about the world token, so we need to generate a dummy one and add
+// it to the inputs for schema matching.
+//
+// Example:
+//   append(int[] list, int el)
+// becomes
+//   append(World w, int[] list, int el)
+//
+// NOTE: The dummy world token has no meaning; the AnnotateEffects pass is
+// necessary to enforce linearization on effectful ops.
+at::optional<std::vector<Value*>> tryMatchMutableSchema(
+    const FunctionSchema& schema,
+    const SourceRange& loc,
+    Graph& graph,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    std::ostream& failure_messages) {
+  JIT_ASSERT(schema.is_mutable);
+  // Add a dummy world token to be matched against
+  const auto worldToken = graph.insertConstant(World());
```

Comment:
Can we verify that all those tokens are gone after annotating the effects?

Reply:
We could probably add a check in `Graph::lint()` that makes sure that there's only a single world token constant